### PR TITLE
TST: Use less autouse=True when unecessary

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

### CODE REVIEW

File: pandas/tests/io/json/test_pandas.py [hunk 1]
```diff
@@ -49,11 +39,18 @@ def assert_json_roundtrip_equal(result, expected, orient):
...
+
+        _frame = DataFrame(_seriesd)
 
-        yield
+        _cat_frame = _frame.copy()
+        cat = ["bah"] * 5 + ["bar"] * 5 + ["baz"] * 5 + ["foo"] * (len(_cat_frame) - 15)
+        _cat_frame.index = pd.CategoricalIndex(cat, name="E")
+        _cat_frame["E"] = list(reversed(cat))
+        _cat_frame["sort"] = np.arange(len(_cat_frame), dtype="int64")
+        return _cat_frame.copy()
```
Comment (jbrockmendel) [inline pandas/tests/io/json/test_pandas.py:53]: i think the copy is unnecessary here

Comment (mroeschke) [reply pandas/tests/io/json/test_pandas.py]: Nice catch. Changed.

File: pandas/tests/io/json/test_pandas.py [hunk 2]
```diff
@@ -49,11 +39,18 @@ def assert_json_roundtrip_equal(result, expected, orient):
...
-    def setup(self):
-        self.categorical = _cat_frame.copy()
+    @pytest.fixture
+    def categorical_frame(self):
+        _seriesd = tm.getSeriesData()
+
+        _frame = DataFrame(_seriesd)
 
-        yield
+        _cat_frame = _frame.copy()
```
Comment (jbrockmendel) [inline pandas/tests/io/json/test_pandas.py:48]: this copy is probably unnecessary too.  ill remove it in my next Assorted Cleanups branch

File: pandas/tests/io/parser/test_textreader.py [hunk 3]
```diff
@@ -25,27 +24,23 @@
 
 
 class TestTextReader:
-    @pytest.fixture(autouse=True)
-    def setup_method(self, datapath):
-        self.dirpath = datapath("io", "parser", "data")
-        csv1_dirpath = datapath("io", "data", "csv")
-        self.csv1 = os.path.join(csv1_dirpath, "test1.csv")
-        self.csv2 = os.path.join(self.dirpath, "test2.csv")
-        self.xls1 = os.path.join(self.dirpath, "test.xls")
```
Comment (jbrockmendel) [inline pandas/tests/io/parser/test_textreader.py:34]: the rest of these aren't used?

File: pandas/tests/io/json/test_pandas.py (hunk 2 above)
Comment (mroeschke) [reply pandas/tests/io/json/test_pandas.py:48]: Thanks 🙏

File: pandas/tests/io/parser/test_textreader.py (hunk 3 above)
Comment (mroeschke) [reply pandas/tests/io/parser/test_textreader.py:34]: Ya I didn't see them used in this file (and the test class isn't inherited AFAICT)

File: pandas/tests/reshape/merge/test_merge_asof.py [hunk 4]
```diff
@@ -163,33 +175,30 @@ def test_examples4(self):
...
-    def test_basic_categorical(self):
+    def test_basic_categorical(self, trades, asof, quotes):
 
-        expected = self.asof
-        trades = self.trades.copy()
+        expected = asof
+        trades = trades.copy()
         trades.ticker = trades.ticker.astype("category")
-        quotes = self.quotes.copy()
+        quotes = quotes.copy()
```
Comment (jbrockmendel) [inline pandas/tests/reshape/merge/test_merge_asof.py:190]: copies likely unnecessary.  NBD.

Comment (mroeschke) [reply pandas/tests/reshape/merge/test_merge_asof.py]: Yup. Removed.

**Comment by jbrockmendel** [comment]
thanks @mroeschke

**Pull request merged**